a10_octavia/controller/worker/tasks/decorators.py

```diff
@@ -63,6 +63,41 @@ def wrapper(self, *args, **kwargs):
     return wrapper
 
 
+def axapi_client_decorator_for_revert(func):
```

Comment:
I agree with this in principle, but there is too much code duplication. We could instead add a switch to throw away ConnectionErrors if they are caught. We don't want to catch & pass on all exceptions unless absolutely necessary because that could obscure critical issues.

```python
def axapi_client_decorator(revert=False):
    def func_wrapper(func):
        def inner_func_wrapper(self, *args, **kwargs):
            ...
            try:
                if use_shared_partition or vthunder.partition_name == 'shared':
                    activate_partition(self.axapi_client, "shared")
                else:
                    if vthunder.partition_name != "shared":
                        activate_partition(self.axapi_client, vthunder.partition_name)
            except ConnectionError as ex:
                if not revert:
                    raise ex
            ...
        return inner_func_wrapper
    return func_wrapper
```